scipy/stats/_multivariate.py

```diff
@@ -165,10 +165,12 @@ def __init__(self, M, cond=None, rcond=None, lower=True,
             raise np.linalg.LinAlgError('singular matrix')
         s_pinv = _pinv_1d(s, eps)
         U = np.multiply(u, np.sqrt(s_pinv))
+        L = np.multiply(u, np.sqrt(np.where(s > eps, s, 0)))
```

Comment:
Why is special treatment of the near-zero eigenvalues needed?

Reply:
To enforce singularity, and to be consistent with the rest of the class. The docstring mentions that `"Singular values smaller than rcond * largest_eigenvalue are considered zero."`. These small eigenvalues are also ignored when computing `self.log_pdet`. 

If such eigenvalues exist, then the matrix is numerically rank deficient. If that is the case and we're trying to generate samples, not rounding them to zero will introduce spurious variance along the corresponding eigenvectors. The magnitude of this variance depends on numerical round-off, and may not be deterministic.

Reply:
@ilayn I think now that your comment referred to this? How should the code be changed?

Reply:
Ah, just saw this. Yes I will take a look during lunch.

Reply:
To retain backward compatibility, there aren't any changes to the `_PSD` class any more.